tests/functional/lang/eval-fail-eol-2.nix

```diff
@@ -0,0 +1,2 @@
+# foo
invalid
```

Comment:
Oof, we treat a single `\r` as a newline? That's pretty odd. I assume this was the behavior already, just untested, so this shouldn't block the PR, but we should consider some follow-up work to fix this.